spec/js/numbers/long_short_decimal.spec.js

```diff
@@ -0,0 +1,80 @@
+const { join, resolve } = require("path");
+
+const rootPath = resolve("twitter_cldr/full");
+const TwitterCldr = require(join(rootPath, "core"));
+
+const testData = {
+  Long: {
+    "en": [
+      // edge cases
+      { raw: 999_499, precisions: ["999 thousand", "999.5 thousand"] },
+      { raw: 999_500, precisions: ["1 million", "999.5 thousand"] },
+      { raw: 1_000_000, precisions: ["1 million", "1.0 million"] },
+
+      // fractional
+      { raw: 999.49, precisions: ["999", "999.5"], default: "999.49" },
+      { raw: 999.5, precisions: ["1 thousand", "999.5"], default: "999.5" },
```

Comment:
I did originally try adding some test cases like `1_000_000.5`, but tbh it's not even clear to me what that should be formatted to at `precision = 1` — `1,000,000.5`? `1.0 million`? Something else entirely? Plus it's enough of a corner case that it's probably not worth worrying about as long as it does something vaguely sane.

Reply:
Generally speaking these formatters are used to format really large numbers so the decimal part isn't significant. Both twitter-cldr-rb and ICU format `1_000_000.5` (precision 1) as "1.0 million". However twitter-cldr-rb outputs "1.0000005 million" when formatted with a precision of 7, which means the decimal isn't entirely discarded. In any case I think these tests are legit.